apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/common/ClaimDao.java

```diff
@@ -244,6 +250,31 @@ Predicate createMbiPredicate(
     return answer;
   }
 
+  /**
+   * Optionally create a {@link Predicate} for the where clause that limits returned entities to
+   * those with a phase number greater than or equal to our {@link ClaimDao#minimumPhaseNumber}.
+   * Only creates the predicate for resources that implement the {@link EntityWithPhaseNumber}
+   * interface.
+   *
+   * @param builder The {@link CriteriaBuilder} being used in current query.
+   * @param root The {@link Root} for the claim in the query
+   * @param resourceType The {@link ResourceTypeV2} that defines properties required for the query.
+   * @return {@link Optional} that is empty if no query needed or filled with a {@link Predicate} if
+   *     one is needed
+   */
+  Optional<Predicate> createPhasePredicate(
+      CriteriaBuilder builder, Path<?> root, ResourceTypeV2<?, ?> resourceType) {
+    var answer = Optional.<Predicate>empty();
+    if (minimumPhaseNumber > 0
+        && EntityWithPhaseNumber.class.isAssignableFrom(resourceType.getEntityClass())) {
+      final var phaseNumber = root.<Short>get(EntityWithPhaseNumber.PhaseFieldName);
```

Comment:
This seems to be the only place that `EntityWithPhaseNumber` is used which doesn't seem to add a huge amount of value for introducing a new abstraction.